docs: Update website to include SIGHUP, SIGTERM, and SIGINT behaviour (#3140)
hashicorp/boundary: website/content/docs/configuration/index.mdx

- [`plugins`](/boundary/docs/configuration/plugins): Configures options for plugins.

## Signals

> On `SIGHUP`, worker and controller processes will reload their configuration file to pick up updated values

Reviewer: I'd add a caveat that not all configuration file values are able to be updated at runtime. Bonus points if you figure out which ones and ensure they're documented as such on the appropriate config pages :-)

Author: Thanks! I'll hunt these down

- [`plugins`](/boundary/docs/configuration/plugins): Configures options for plugins.

## Signals

> On `SIGHUP`, worker and controller processes will reload their configuration file to pick up updated values

Reviewer 2: ```suggestion
The `SIGHUP` signal causes worker and controller processes to reload their configuration files to pick up updated values.
```

> ~> **Note:** Not all configuration values can be reloaded. Please see the configuration pages for
> [workers](/boundary/docs/configuration/worker/) and [controllers](/boundary/docs/configuration/controller) for details
> on which values can be reloaded on `SIGHUP`.

Reviewer 2: ```suggestion
~> **Note:** You cannot reload all configuration values using the `SIGHUP` signal. Refer to the configuration pages for
[workers](/boundary/docs/configuration/worker/) and [controllers](/boundary/docs/configuration/controller) for details
on which values can be reloaded on `SIGHUP`.
```

Author: thanks for all these!!

> On `SIGTERM` or `SIGINT`, worker and controller processes will enter graceful shutdown. Controller graceful shutdown
> will close listeners and servers before shutting down. Worker graceful shutdown will wait for sessions to drain
> before shutting down. Workers in a graceful shutdown state will not receive new work, including session proxying, from the control plane.

Reviewer 2: ```suggestion
The `SIGTERM` and `SIGINT` signals cause worker and controller processes to enter graceful shutdown. A graceful shutdown for a controller closes listeners and servers before shutting down the controller. A graceful shutdown for a worker waits for any sessions to drain
before shutting down the worker. Workers in a graceful shutdown state do not receive any new work, including session proxying, from the control plane.
```
